braintree/util/http.py

```diff
@@ -31,7 +31,7 @@ class ContentType(object):
 
     @staticmethod
     def is_error_status(status):
-        return status not in [200, 201, 422]
+        return status not in [200, 201, 204, 422]
```

Comment:
`.delete()` on apple pay domains issue a 204, which is a valid success HTTP code.